homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -19,14 +19,21 @@ class YaleDataUpdateCoordinator(DataUpdateCoordinator):
     def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize the Yale hub."""
         self.entry = entry
-        self.yale: YaleSmartAlarmClient | None = None
+        self.yale: YaleSmartAlarmClient = None
         super().__init__(
             hass,
             LOGGER,
             name=DOMAIN,
             update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
         )
 
+    def get_client(self) -> YaleSmartAlarmClient:
+        """Connect to Yale API."""
+        self.yale = YaleSmartAlarmClient(
+            self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
+        )
+        return self.yale
```

Comment:
Why return an instance attribute from an instance method?

Reply:
Needed to return something for setup_entry to know it was succesful but changed it to bool now instead

Reply:
Which I can't upload until the previous issue is resolved